src/charm.py

```diff
@@ -1,34 +1,146 @@
 #!/usr/bin/env python3
 # Copyright 2021 Canonical Ltd.
 # See LICENSE file for licensing details.
-
+import yaml
+import json
 import logging
+import ops.charm
 import subprocess
+from subprocess import check_call
 from urllib.request import urlopen
 
-from ops.charm import CharmBase
 from ops.main import main
-from ops.model import BlockedStatus, MaintenanceStatus, WaitingStatus
+from ops.model import BlockedStatus, MaintenanceStatus, WaitingStatus, ActiveStatus, Relation
 from charms.operator_libs_linux.v0 import apt
+from charms.operator_libs_linux.v0.systemd import service_restart, service_resume
+from mongoserver import MongoDB, MONGODB_PORT
+from typing import Optional
 
 logger = logging.getLogger(__name__)
 
+PEER = "mongodb"
+
 
-class MongodbOperatorCharm(CharmBase):
+class MongodbOperatorCharm(ops.charm.CharmBase):
     """Charm the service."""
 
     def __init__(self, *args):
         super().__init__(*args)
+        self.port = MONGODB_PORT
+
         self.framework.observe(self.on.install, self._on_install)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.start, self._on_start)
 
-    def _on_install(self, _):
-        """Install prerequisites for the application"""
+    def _on_install(self, _: ops.charm.InstallEvent) -> None:
+        """Handle the install event (fired on startup)
+        Handles the startup install event -- installs updates the apt cache, installs mongoDB
+        Args:
+            _ (ops.charm.InstallEvent): The install event that fired.
+        Returns:
+            None: None
+        """
         self.unit.status = MaintenanceStatus("installing MongoDB")
         self._add_mongodb_org_repository()
         self._install_apt_packages(["mongodb-org"])
-        self.unit.status = WaitingStatus("Waiting to start MongoDB")
+        try:
+            mongo_path = subprocess.check_output(["which", "mongod"]).decode().strip()
+            if not mongo_path == "/usr/bin/mongod":
+                logger.error("wrong mongodb path found %s want /usr/bin/mongod", mongo_path)
```

Comment:
```suggestion
                logger.error("wrong mongodb path found: '%s', expecting '/usr/bin/mongod'", mongo_path)
```

Reply:
removed this as it was an artifact from something else